docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+
+| Items parameter | Description                                                                                  | Type     |
+| --------------- | -------------------------------------------------------------------------------------------- | -------- |
+| `name`          | Name of the setting item<br/>Must use the `options.setting-name` format.                     | `String` |
+| `description`   | Description of the setting item to use in the Content-type Builder                           | `String` |
+| `intlLabel`     | Translation for the abel of the setting item                                                 | [`IntlObject`](https://formatjs.io/docs/react-intl/) |
+| `type`          | Type of setting (see [Setting types and specific options](#))                                 | `String` |
+
+<!-- TODO: add a super simple example with one base setting, one advanced setting, and a simple validator -->
+::: details Example: Declaring options for an example "color" custom field:
+
+```jsx
+// path: ./src/plugins/my-custom-field-plugin/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    // …
+    options: {
+      base: [
+        {
+          sectionTitle: {
+            id: 'color-picker.color.section.format',
+            defaultMessage: 'Format',
+          },
+          items: [
+            {
+              intlLabel: {
+                id: 'color-picker.color.format.label',
+                defaultMessage: 'Color format',
+              },
+              name: 'options.format',
+              type: 'select',
+              value: 'hex',
+              options: [
+                {
+                  key: 'hex',
+                  value: 'hex',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.hex',
+                      defaultMessage: 'Hexadecimal',
+                    },
+                  },
+                },
+                {
+                  key: 'rgba',
+                  value: 'rgba',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.rgba',
+                      defaultMessage: 'RGBA',
+                    },
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      advanced: [],
+      validator: (args) => ({
+        'color-picker': yup.object().shape({
+          format: yup.string().oneOf(['hex', 'rgba']),
+        }),
+      }),
+    },
+  });
+}
+```
+
+:::
+
+:::note
+When extending a custom field’s base and advanced forms in the Content-type Builder, it is not currently possible to import custom input components.
```

Comment:
Technically it is possible but the API is not documented, I think it was an internal API built for i18n, you can see it used [here](https://github.com/strapi/strapi/blob/74c33527616889fdec2ff1d4fec7ecbea457f5be/packages/plugins/i18n/admin/src/index.js#L90) and then how the advanced form uses the component [here](https://github.com/strapi/strapi/blob/74c33527616889fdec2ff1d4fec7ecbea457f5be/packages/plugins/i18n/admin/src/index.js#L107)